2025-05-30T21:16:15.2126126Z TRACE:: Engine.Main(): LEAN ALGORITHMIC TRADING ENGINE v2.5.0.0 Mode: DEBUG (64bit) Host: ubuntu-dev
2025-05-30T21:16:15.2165050Z TRACE:: Engine.Main(): Started 9:16 PM
2025-05-30T21:16:15.2197423Z TRACE:: Config.Get(): Configuration key not found. Key: lean-manager-type - Using default value: LocalLeanManager
2025-05-30T21:16:15.2352411Z TRACE:: JobQueue.NextJob(): Selected /LeanCLI/main.py
2025-05-30T21:16:15.2646102Z TRACE:: Config.GetValue(): scheduled-event-leaky-bucket-capacity - Using default value: 120
2025-05-30T21:16:15.2646485Z TRACE:: Config.GetValue(): scheduled-event-leaky-bucket-time-interval-minutes - Using default value: 1440
2025-05-30T21:16:15.2646574Z TRACE:: Config.GetValue(): scheduled-event-leaky-bucket-refill-amount - Using default value: 18
2025-05-30T21:16:15.2650603Z TRACE:: Config.GetValue(): storage-limit - Using default value: 10737418240
2025-05-30T21:16:15.2650783Z TRACE:: Config.GetValue(): storage-permissions - Using default value: 3
2025-05-30T21:16:15.2651179Z TRACE:: Config.Get(): Configuration key not found. Key: optimization-id - Using default value: 
2025-05-30T21:16:15.2651270Z TRACE:: Config.Get(): Configuration key not found. Key: backtest-name - Using default value: local
2025-05-30T21:16:15.2676817Z TRACE:: Config.Get(): Configuration key not found. Key: data-channel - Using default value: 
2025-05-30T21:16:15.2677777Z TRACE:: Config.Get(): Configuration key not found. Key: python-venv - Using default value: 
2025-05-30T21:16:15.2678267Z TRACE:: Config.Get(): Configuration key not found. Key: out-of-sample-max-end-date - Using default value: 
2025-05-30T21:16:15.2678361Z TRACE:: Config.GetValue(): out-of-sample-days - Using default value: 0
2025-05-30T21:16:15.2683775Z TRACE:: Config.Get(): Configuration key not found. Key: data-permission-manager - Using default value: DataPermissionManager
2025-05-30T21:16:15.2683914Z TRACE:: Config.Get(): Configuration key not found. Key: data-monitor - Using default value: QuantConnect.Data.DataMonitor
2025-05-30T21:16:15.2774686Z TRACE:: Config.GetValue(): algorithm-creation-timeout - Using default value: 90
2025-05-30T21:16:15.2833212Z TRACE:: Config.GetValue(): zip-data-cache-provider - Using default value: 10
2025-05-30T21:16:15.2838920Z TRACE:: Config.Get(): Configuration key not found. Key: fundamental-data-provider - Using default value: CoarseFundamentalDataProvider
2025-05-30T21:16:15.2851532Z TRACE:: AlgorithmManager.CreateTokenBucket(): Initializing LeakyBucket: Capacity: 120 RefillAmount: 18 TimeInterval: 1440
2025-05-30T21:16:15.2853828Z TRACE:: Config.GetValue(): algorithm-manager-time-loop-maximum - Using default value: 20
2025-05-30T21:16:15.2888085Z TRACE:: Config.GetValue(): downloader-data-update-period - Using default value: 7
2025-05-30T21:16:15.2893652Z TRACE:: Engine.Run(): Resource limits '0' CPUs. 2147483647 MB RAM.
2025-05-30T21:16:15.2896421Z TRACE:: TextSubscriptionDataSourceReader.SetCacheSize(): Setting cache size to 71582788 items
2025-05-30T21:16:15.6808610Z TRACE:: PythonInitializer.Initialize(): start...
2025-05-30T21:16:16.2978109Z TRACE:: PythonInitializer.Initialize(): ended
2025-05-30T21:16:16.2988173Z TRACE:: AlgorithmPythonWrapper(): Python version 3.11.8 | packaged by conda-forge | (main, Feb 16 2024, 21:14:50) [GCC 12.3.0]: Importing python module main
2025-05-30T21:16:17.8845593Z TRACE:: AlgorithmPythonWrapper(): main successfully imported.
2025-05-30T21:16:17.8877946Z TRACE:: AlgorithmPythonWrapper(): Creating IAlgorithm instance.
2025-05-30T21:16:17.8959493Z TRACE:: Config.Get(): Configuration key not found. Key: databases-refresh-period - Using default value: 1.00:00:00
2025-05-30T21:16:17.9771300Z TRACE:: Config.GetValue(): mute-python-library-logging - Using default value: True
2025-05-30T21:16:17.9805749Z TRACE:: LocalObjectStore.Initialize(): Storage Root: /Storage. StorageFileCount 9999999. StorageLimit 10240MB
2025-05-30T21:16:17.9843647Z TRACE:: Config.GetValue(): zip-data-cache-provider - Using default value: 10
2025-05-30T21:16:17.9871374Z TRACE:: HistoryProviderManager.Initialize(): history providers [SubscriptionDataReaderHistoryProvider]
2025-05-30T21:16:17.9892063Z TRACE:: BaseSetupHandler.Setup(LocalPlatform): UID: 357130, PID: -430634829, Version: 2.5.0.0, Source: WebIDE
2025-05-30T21:16:17.9912523Z TRACE:: Config.Get(): Configuration key not found. Key: security-data-feeds - Using default value: 
2025-05-30T21:16:18.0007692Z TRACE:: BacktestingResultHandler(): Sample Period Set: 525.96
2025-05-30T21:16:18.0017933Z TRACE:: Config.GetValue(): forward-console-messages - Using default value: True
2025-05-30T21:16:18.0264020Z ERROR:: Engine.Run(): During the algorithm initialization, the following exception has occurred: error return without exception set
  at Initialize
    self.securities = {}
    ^^^^^^^^^^^^^^^
 in main.py: line 11
 error return without exception set
2025-05-30T21:16:18.0267631Z TRACE:: JOB HANDLERS:
         DataFeed:             QuantConnect.Lean.Engine.DataFeeds.FileSystemDataFeed
         Setup:                QuantConnect.Lean.Engine.Setup.BacktestingSetupHandler
         RealTime:             QuantConnect.Lean.Engine.RealTime.BacktestingRealTimeHandler
         Results:              QuantConnect.Lean.Engine.Results.BacktestingResultHandler
         Transactions:         QuantConnect.Lean.Engine.TransactionHandlers.BacktestingTransactionHandler
         Object Store:         QuantConnect.Lean.Engine.Storage.LocalObjectStore
         History Provider:     QuantConnect.Lean.Engine.HistoricalData.HistoryProviderManager
         Brokerage:            QuantConnect.Brokerages.Backtesting.BacktestingBrokerage
         Data Provider:        QuantConnect.Lean.Engine.DataFeeds.DefaultDataProvider

2025-05-30T21:16:18.0270210Z TRACE:: FileSystemDataFeed.Exit(): Start. Setting cancellation token...
2025-05-30T21:16:18.0276587Z TRACE:: FileSystemDataFeed.Exit(): Exit Finished.
2025-05-30T21:16:18.0278999Z TRACE:: BacktestingResultHandler.Exit(): starting...
2025-05-30T21:16:18.0292716Z ERROR:: During the algorithm initialization, the following exception has occurred: error return without exception set
  at Initialize
    self.securities = {}
    ^^^^^^^^^^^^^^^
 in main.py: line 11
 error return without exception set
 error return without exception set
  at Initialize
    self.securities = {}
    ^^^^^^^^^^^^^^^
 in main.py: line 11
 error return without exception set
2025-05-30T21:16:18.0335812Z TRACE:: BacktestingResultHandler.Exit(): Saving logs...
2025-05-30T21:16:18.0346593Z TRACE:: Debug: Your log was successfully created and can be retrieved from: /Results/1653806800-log.txt
2025-05-30T21:16:18.0346741Z TRACE:: BacktestingResultHandler.Run(): Ending Thread...
2025-05-30T21:16:18.0348235Z TRACE:: StopSafely(): Waiting for 'Result Thread' thread to stop...
2025-05-30T21:16:18.0887929Z TRACE:: 
STATISTICS:: OrderListHash d41d8cd98f00b204e9800998ecf8427e
2025-05-30T21:16:18.0888122Z TRACE:: BacktestingResultHandler.SendAnalysisResult(): Processed final packet
2025-05-30T21:16:18.0889617Z TRACE:: Engine.Run(): Disconnecting from brokerage...
2025-05-30T21:16:18.0889999Z TRACE:: Engine.Run(): Disposing of setup handler...
2025-05-30T21:16:18.0890255Z TRACE:: Engine.Main(): Analysis Completed and Results Posted.
2025-05-30T21:16:18.0914700Z TRACE:: Engine.Main(): Packet removed from queue: 1653806800
2025-05-30T21:16:18.0915354Z TRACE:: LeanEngineSystemHandlers.Dispose(): start...
2025-05-30T21:16:18.0919727Z TRACE:: LeanEngineSystemHandlers.Dispose(): Disposed of system handlers.
2025-05-30T21:16:18.0922438Z TRACE:: LeanEngineAlgorithmHandlers.Dispose(): start...
2025-05-30T21:16:18.0926713Z TRACE:: LeanEngineAlgorithmHandlers.Dispose(): Disposed of algorithm handlers.
